runtime/registry.go

```diff
@@ -16,21 +16,56 @@ package runtime
 
 import (
 	"fmt"
+	"path/filepath"
+
+	"github.com/alibaba/sealer/logger"
+
+	"github.com/alibaba/sealer/utils"
+
+	"github.com/alibaba/sealer/common"
 )
 
-func getRegistryHost(ip string) (host string) {
-	return fmt.Sprintf("%s %s", ip, SeaHub)
+type RegistryConfig struct {
+	IP     string
+	Domain string
+	Volume string
+}
+
+func (d *Default) getRegistryHost() (host string) {
+	cf := d.getRegistryConfig()
+	if cf == nil {
+		return fmt.Sprintf("%s %s", d.Masters[0], SeaHub)
+	}
+	return fmt.Sprintf("%s %s", cf.IP, cf.Domain)
+}
+
+func (d *Default) getRegistryConfig() *RegistryConfig {
+	var config RegistryConfig
+	registryConfigPath := filepath.Join(common.DefaultClusterBaseDir(d.ClusterName), "etc/registry.yaml")
+	if utils.IsFileExist(registryConfigPath) {
+		err := utils.UnmarshalYamlFile(registryConfigPath, &config)
+		if err == nil && config.IP != "" && config.Domain != "" {
```

Comment:
I think we can set default domain and default host IP here.

Reply:
default host IP defined in the getregistryhost() function. so, I don't think we need to change it.